api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/BillServiceClientIntegrationTest.java

```diff
@@ -369,5 +370,89 @@ void shouldUpdateSpecificFieldsOfBill() throws Exception {
                 .verifyComplete();
     }
 
+    @Test
+    void getNonExistentBillById() {
+        server.enqueue(new MockResponse().setResponseCode(404));
+
+        Mono<BillResponseDTO> billResponseDTOMono = billServiceClient.getBilling("nonexistentId");
+
+        StepVerifier.create(billResponseDTOMono)
+                .expectError(WebClientResponseException.NotFound.class)
+                .verify();
+    }
+
+    @Test
+    void getBillByInvalidVetId() {
+        server.enqueue(new MockResponse().setResponseCode(400));
+
+        Flux<BillResponseDTO> billResponseDTOMono = billServiceClient.getBillsByVetId("invalidVetId");
+
+        StepVerifier.create(billResponseDTOMono)
+                .expectError(WebClientResponseException.BadRequest.class)
+                .verify();
+    }
+
+    @Test
+    void getBillsByInvalidCustomerId() {
+        server.enqueue(new MockResponse().setResponseCode(500));
+
+        Flux<BillResponseDTO> billResponseDTOMono = billServiceClient.getBillsByOwnerId("invalidCustomerId");
+
+        StepVerifier.create(billResponseDTOMono)
+                .expectError(WebClientResponseException.InternalServerError.class)
+                .verify();
+    }
```

Comment:
These two tests seem extremely similar. Should they be throwing different response codes?

Reply:
Yes, thank  you for catching that I have updated it to excpect 400 as it should.